dbms/src/DataStreams/WindowBlockInputStream.h

```diff
@@ -71,6 +71,16 @@ struct RowNumber
 /* Implementation details.*/
 struct WindowTransformAction
 {
+private:
+    // Used for calculating the frame start
+    RowNumber stepToFrameStart(const RowNumber & current_row, UInt64 n);
```

Comment:
Why not just use `const RowNumber & current_row, WindowFrame & frame` as the argument?

Reply:
> WindowFrame

okk